alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -50,7 +53,46 @@ void RunnableObject::start(RunningPlan* rp)
     }
     ++_signalState;
     _signalContext.store(rp);
-    _engine->editScheduler().schedule(std::bind(&RunnableObject::doInit, this));
+    if(_requiresParameters) {
+        assert(rp->getParent());
+        assert(rp->getParent()->getBasicPlan());
+        const auto& wrappers = rp->getParent()->getActiveState()->getConfAbstractPlanWrappers();
+        auto it = std::find_if(wrappers.begin(), wrappers.end(), [this](const auto& wrapper_ptr){
+            return wrapper_ptr->getAbstractPlan()->getName() == _name;
+        });
+        assert(it != wrappers.end());
+
+        int64_t wrapperId = (*it)->getId();
+
+        std::shared_ptr<BlackBoard> parentBlackBoard = rp->getParent()->getBasicPlan()->getBlackBoard();
+        auto& planAttachment = rp->getParent()->getBasicPlan()->getPlanAttachment(wrapperId);
+        auto initCall = [&](){
+            _blackBoard->clear();
+            if(!planAttachment->setParameters(*parentBlackBoard, *_blackBoard)) {
+                std::cerr << "Setting parameters failed, supposedly as the context has already changed.  Plan will not be scheduled" << std::endl;
+                return;
+            }
+            doInit();
+        };
+        _engine->editScheduler().schedule(initCall);
+    } else {
+
+        std::shared_ptr<BlackBoard> parentBlackBoard;
+        if(rp->getParent() && rp->getParent()->getBasicPlan()) {
+            parentBlackBoard = rp->getParent()->getBasicPlan()->getBlackBoard();
+
+        }
+        auto initCall = [this, parentBlackBoard=std::move(parentBlackBoard)](){
+            // Share blackboard with parent if we have one, or start fresh otherwise
+            if(parentBlackBoard) {
+                _blackBoard = parentBlackBoard;
+            } else {
+                _blackBoard = std::make_shared<BlackBoard>();
```

Comment:
I can't think of a case where the parent would not have a blackboard of its own. In which case we should rather assert here.

Reply:
If this is master plan, there is no parent blackboard.  This is just to avoid having separate lamba call for master plan

Reply:
makes sense.